cloud/deployment/status.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2023 Terramate GmbH
+// SPDX-License-Identifier: MPL-2.0
+
+// Package deployment provides types and helpers for cloud deployments.
+package deployment
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/bits"
+	"strconv"
+
+	"github.com/terramate-io/terramate/errors"
+)
+
+const (
+	// ErrInvalidStatus represents an invalid deployment status error.
+	ErrInvalidStatus errors.Kind = "invalid deployment status"
+)
+
+// Status of a stack or deployment
+type Status uint8
+
+const (
+	OK       Status = 1 << iota // OK status is used when the stack ran successfully.
```

Comment:
My kingdom for sum types in golang, good lord. This file is an excellent solution/work-around, kudos.